providers/implementations/exchange/dh_exch.c

```diff
@@ -271,12 +270,12 @@ static void *dh_dupctx(void *vpdhctx)
     dstctx->kdf_ukm = NULL;
     dstctx->kdf_cekalg = NULL;
 
-    if (dstctx->dh != NULL && !DH_up_ref(srcctx->dh))
+    if (srcctx->dh != NULL && !DH_up_ref(srcctx->dh))
         goto err;
     else
         dstctx->dh = srcctx->dh;
```

Comment:
I wonder why there's an `else` branch...  it kinda made me go "huh?"

Wouldn't this be a little bit clearer?

``` C
    if (srcctx->dh != NULL && !DH_up_ref(srcctx->dh))
        goto err;
    dstctx->dh = srcctx->dh;
```

Reply:
I purposefully made this just a comment, not a change request.  I'm fine with this going in as is...